src/rules/padding.ts

```diff
@@ -77,18 +77,39 @@ interface PaddingContext {
 }
 
 // Creates a StatementTester to test an ExpressionStatement's first token name
-const createTokenTester = (tokenName: string): StatementTester => {
+const createTokenTester = (
+  tokenName: string,
+  expectedNodeType = 'ExpressionStatement',
+): StatementTester => {
   return (node: Node, sourceCode: SourceCode): boolean => {
     const token = sourceCode.getFirstToken(node);
 
     return (
-      node.type === 'ExpressionStatement' &&
+      node.type === expectedNodeType &&
       token.type === 'Identifier' &&
       token.value === tokenName
     );
   };
 };
 
+// Creates a StatementTester which digs into awaited ExpressionStatements
+const createAwaitedTokenTester = (tokenName: string): StatementTester => {
+  return (node: Node, sourceCode: SourceCode): boolean => {
+    let activeNode = node;
+    let expectedType = 'ExpressionStatement';
+
+    if (
+      activeNode.type === 'ExpressionStatement' &&
+      activeNode.expression.type === 'AwaitExpression'
+    ) {
+      activeNode = activeNode.expression.argument;
+      expectedType = 'CallExpression';
```

Comment:
What is this being used for? I wasn't able to find `CallExpression` anywhere in the code.

Reply:
It's the value of `node.expression.argument.type` when `node.expression.type` is `'AwaitExpression'`. It gets passed up to `createTokenTester()` and used to verify that the `node` there is the type that we expect it to be.

Sadly I can't really give a better explanation than that - this is just what I discovered by stepping through the code as it was running into an expression like `await expect(...)...`.

Reply:
re: `CallExpression` 

I think that we'll always get a `CallExpression` in our particular case with `await expect(...)`, _but_ there's definitely more to `AwaitExpression`'s `argument` than that. Here's the interface for an [`AwaitExpression`](https://github.com/estree/estree/blob/master/es2017.md). The `argument` is an [`Expression`](https://github.com/estree/estree/blob/master/es5.md#expressions), such as `CallExpression`, `MemberExpression`, etc.

@tmercswims How would you feel about handling `AwaitExpression`s more generically in the original `createTokenTester`? Then we wouldn't have this additional tester just for expect. Maybe something like...

```
# pseudo-code

if expression statement
  if await expression
    node = node.expression.argument
  end
  
  token = get first token from node
  return token is identifier and is a match for token value
else
  return false
```

I think _something_ like that above would provide a more general fix, because someone could always do something like this...

```js
async function whatHaveIDone() {
  // These describes are converted to resolved promises and are a valid usage of await...
  await describe("why have i done this", () => {});
  await describe("i am a monster", () => {});
}

whatHaveIDone();
```

This would mean there's no particular check for `CallExpression` although that is almost always what we would get. We'd just essentially be saying "Oh, it's an await... I need to look at the next node for the first token" in all cases, not just for expect.

Reply:
Absolutely makes sense, and something I probably should have thought to refine 😅 I've done the change pretty much exactly how you describe, and it works great.